astropy/utils/data.py

```diff
@@ -232,6 +233,7 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
         try:
             import bz2
         except ImportError:
+            fileobj.close()
```

Comment:
If `name_or_obj` is passed in as a fileobj (rather than a string path), then we shouldn't be closing the file object on the user's behalf, since that could result in closing the same file twice.  Instead, do:

```
for fd in close_fds:
    fd.close()
```